5-April-22/upgrade.md

```diff
@@ -0,0 +1,94 @@
+# Juno April 5, 2022 Chain Halt Upgrade
+This upgrade is effectively a hard fork with the same chain-id. What this means, is we'll need to delete all previous data and start from a new genesis.
+
+NOTE: This assumes you've already ran through setting up a juno node here: https://docs.junonetwork.io/validators/getting-setup
+
+## Performing the  Upgrade
+
+### 1. Stop the node
+```
+sudo systemctl stop junod
+```
+
+### 2. Backup ~/.juno/config/priv_validator_key.json
+This is only relevant if you're upgrading your validator. Your priv_validator_key.json is how your validator is identified. If you haven't backed it up already, *DO SO NOW*.
+
+An example method for doing so is as follows, which will copy it to the home dir:
+```sh
+cd ~
+cp ~/.juno/config/priv_validato_state.json .
+```
+
+### 3. Delete previous chain state
+Because we'll be starting from a new genesis, the previous data is no longer necessary. Similarly, the priv_validator_state.json is not necessary because there is no previous block to sign from.
+
+```sh
+junod unsafe-reset-all
+```
+
+### 4. Download and install the new binary
+```
+git clone https://github.com/CosmosContracts/juno
+cd juno
+git fetch
+git checkout v3.0.0
+make install
+```
+
+To confirm the verison, do:
+```
+junod version --long
+```
+
+(TEMP - NEEDS TO BE REPLACED)
+
+```
+name: juno
+server_name: junod
+version: v2.1.0
+commit: e6b8c212b178cf575035065b78309aed547b1335
+build_tags: netgo muslc,
+go: go version go1.17.3 linux/amd64
+```
+
+### 5. Download the new genesis
+```sh:
+curl TEMP_NEED_URL_PATH > ~/.juno/config/genesis.json
+```
+
```

Comment:
ahould we be clearing peers and confirming peers for upgraded nodes only?

should we instruct to delete address book?

Reply:
if using cosmovisor, need to delete the contents of upgrades dir, replace genesis bin with 3.0.0 and delete current symlink.

Reply:
> if using cosmovisor, need to delete the contents of upgrades dir, replace genesis bin with 3.0.0 and delete current symlink.

Yes, many are using that, we should cover those cases.

Reply:
I believe there is no need to delete the address book (exclude you just want to cleanup your peers but its not mandatory)

Reply:
Deleting it and purging peers will help starting back up and avoid connecting to nodes that can no longer be dialed. 👍

Reply:
i have some new Cosmovisor install instructions let me know if you need them left anywhere.